server/jamovi/server/server.py

```diff
@@ -141,6 +141,76 @@ def get(self, instance_id, analysis_id, path):
             self.set_header('Cache-Control', 'private, no-cache, must-revalidate, max-age=0')
             self.write(content)
 
+class AppI18nDescriptor(RequestHandler):
+    def get(self, code):
+        content = None
+        try:
+            try:
+                client_path = conf.get('client_path')
+                defn_path = os.path.join(client_path, 'i18n', code + '.json')
+                with open(defn_path, 'rb') as file:
+                    content = file.read()
+            except (KeyError, FileNotFoundError):
+                raise
+            except Exception as e:
+                log.exception(e)
+        except Exception as e:
+            self.set_status(404)
+            self.write('<h1>404</h1>')
+            self.write(str(e))
+        else:
+            self.set_header('Content-Type', 'application/json')
+            self.set_header('Cache-Control', 'private, no-store, must-revalidate, max-age=0')
+            self.write(content)
+
+class AppI18nCodes(RequestHandler):
+    def get(self):
+        list = [ ]
+        client_path = conf.get('client_path')
+        for direntry in os.scandir(os.path.join(client_path, 'i18n') + '/'):
+            if direntry.name.endswith('.json'):
+                list.append(direntry.name[:-5])
+
+        self.set_header('Content-Type', 'application/json')
+        self.set_header('Cache-Control', 'private, no-store, must-revalidate, max-age=0')
+        self.write(json.dumps(list))
+
+class ModuleI18nCodes(RequestHandler):
+    def get(self, module_name):
+
+        list = [ ]
+        module_path = Modules.instance().get(module_name).path
+        i18n_path = os.path.join(module_path, 'i18n') + '/'
+        if os.path.exists(i18n_path):
+            for direntry in os.scandir(os.path.join(module_path, 'i18n') + '/'):
+                list.append(direntry.name[:-5])
+
+        self.set_header('Content-Type', 'application/json')
+        self.set_header('Cache-Control', 'private, no-store, must-revalidate, max-age=0')
+        self.write(json.dumps(list))
+
+class ModuleI18nDescriptor(RequestHandler):
+
+    def get(self, module_name, code):
+        content = None
+        try:
+            try:
+                module_path = Modules.instance().get(module_name).path
+                defn_path = os.path.join(module_path, 'i18n', code + '.json')
+                with open(defn_path, 'rb') as file:
+                    content = file.read()
+            except (KeyError, FileNotFoundError):
+                raise
+            except Exception as e:
+                log.exception(e)
+        except Exception as e:
+            self.set_status(404)
+            self.write('<h1>404</h1>')
+            self.write(str(e))
+        else:
+            self.set_header('Content-Type', 'application/json')
+            self.set_header('Cache-Control', 'private, no-store, must-revalidate, max-age=0')
+            self.write(content)
```

Comment:
can we generate these files at compile time, and then serve them up as static files?